spec/request/mhv_accounts_request_spec.rb

```diff
@@ -29,135 +29,261 @@
+  shared_context 'an unsuccessful POST #create' do
+    it 'fails to create' do
+      VCR.use_cassette('mhv_account_creation/account_creation_unknown_error') do
+        post v0_mhv_account_path
         expect(response).to have_http_status(:bad_request)
+        expect(JSON.parse(response.body)['errors'].first['detail'])
+          .to eq('Something went wrong. Please try again later.')
       end
+    end
+  end
 
-      it 'handles upgrade error in POST #create' do
-        VCR.use_cassette('mhv_account_creation/creates_an_account') do
-          VCR.use_cassette('mhv_account_creation/account_upgrade_unknown_error') do
-            post v0_mhv_account_path
-          end
-        end
+  shared_context 'an unsuccessful POST #upgrade' do
+    it 'fails to upgrade' do
+      VCR.use_cassette('mhv_account_creation/account_creation_unknown_error') do
+        post '/v0/mhv_account/upgrade'
         expect(response).to have_http_status(:bad_request)
+        expect(JSON.parse(response.body)['errors'].first['detail'])
+          .to eq('Something went wrong. Please try again later.')
       end
     end
+  end
+
+  context 'without T&C acceptance' do
+    it_behaves_like 'a successful GET #show', account_state: 'needs_terms_acceptance', account_level: nil
+    it_behaves_like 'a failed POST #create', http_status: :forbidden,
+                                             message: V0::MhvAccountsController::CREATE_ERROR
+    it_behaves_like 'a failed POST #upgrade', http_status: :forbidden,
+                                              message: V0::MhvAccountsController::UPGRADE_ERROR
+    it_behaves_like 'ssn mismatch'
+    it_behaves_like 'non va patient'
+  end
 
-    context 'with an account' do
+  context 'with T&C acceptance' do
+    before { create(:terms_and_conditions_acceptance, terms_and_conditions: terms, user_uuid: user.uuid) }
+
+    it_behaves_like 'ssn mismatch'
+    it_behaves_like 'non va patient'
+
+    context 'without an account' do
+      around(:each) do |example|
+        VCR.use_cassette('mhv_account_type_service/advanced') do
+          example.run
+        end
+      end
+
+      it_behaves_like 'a successful GET #show', account_state: 'no_account', account_level: nil
+      it_behaves_like 'a successful POST #create'
+      it_behaves_like 'a failed POST #upgrade', http_status: :forbidden,
+                                                message: V0::MhvAccountsController::UPGRADE_ERROR
+    end
+
+    context 'with account' do
       let(:mhv_ids) { ['14221465'] }
 
-      context 'that is existing' do
-        it 'responds to GET #show' do
-          get v0_mhv_account_path
-          expect(response).to be_success
-          expect(JSON.parse(response.body)['data']['attributes']['account_state']).to eq('existing')
+      context 'that is' do
+        %w[Basic Advanced].each do |type|
+          context type do
+            around(:each) do |example|
+              VCR.use_cassette('mhv_account_type_service/premium') do
```

Comment:
What's the purpose of wrapping this around the cassette below?

Reply:
Basic and advanced need to get upgraded to Premium. I want the example to return 'Premium' after upgrade is successful.